client/src/components/Signup.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from "react";
+import { UikInput, UikButton } from "../@uik";
+import "../@uik/styles.css";
+import axios from "axios";
+import styled from "styled-components";
+
+const Container = styled.div`
+  display: flex;
+  width: 400px;
+  margin: 0 100px;
+  flex-direction: column;
+`;
+
+class Signup extends Component {
+  state = {
+    username: "",
+    password: "",
+    department: ""
+  };
+
+  handleSubmit = event => {
+    event.preventDefault();
+
+    const endpoint = "http://localhost:3000/api/register";
+    axios
+      .post(endpoint, this.state)
+      .then(res => {
+        console.log(res.data);
+        localStorage.setItem("jwt", res.data.token);
+      })
+      .catch(err => {
+        console.log("ERROR");
+      });
+
+    this.props.history.push("/users");
+  };
+
+  handleInput = event => {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  render() {
+    return (
+      <Container>
+        <form onSubmit={this.handleSubmit}>
+          <div>
+            <UikInput
+              label="Username"
+              name="username"
+              placeholder="your@email.com"
+              value={this.state.username}
```

Comment:
It is a good idea to have a fourth input field that verifies the password, for UX purposes. This can be done solely on the front-end by setting both passwords to state and checking to make sure that they match.